lib/goldiloader.rb

```diff
@@ -8,3 +8,37 @@
 require 'goldiloader/association_options'
 require 'goldiloader/association_loader'
 require 'goldiloader/active_record_patches'
+
+module Goldiloader
+  @globally_enabled = true
+
+  class << self
+
+    # Sets the process-wide enabled status
+    attr_reader :globally_enabled
+
+    def enabled?
+      Thread.current.fetch(:goldiloader_enabled, @enabled)
```

Comment:
This isn't defaulting properly to `globally_enabled`. Can we add test coverage for this too?

```suggestion
      Thread.current.fetch(:goldiloader_enabled, globally_enabled)
```